x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/get_columns.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { EuiBasicTableColumn } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ValuesType } from 'utility-types';
+import {
+  asMillisecondDuration,
+  asPercent,
+  asTransactionRate,
+} from '../../../../../common/utils/formatters';
+import { APIReturnType } from '../../../../services/rest/createCallApmApi';
+import { px, unit } from '../../../../style/variables';
+import { SparkPlot } from '../../../shared/charts/spark_plot';
+import { ImpactBar } from '../../../shared/ImpactBar';
+import { TransactionDetailLink } from '../../../shared/Links/apm/transaction_detail_link';
+import { TruncateWithTooltip } from '../../../shared/truncate_with_tooltip';
+import { TransactionGroupsOverview } from './';
+
+export type ServiceTransactionGroupItem = ValuesType<
+  TransactionGroupsOverview['transactionGroups']
+>;
+type TransactionGroupMetrics = APIReturnType<'GET /api/apm/services/{serviceName}/transactions/groups/agg_results'>;
+
+function getLatencyAggregationTypeLabel(latencyAggregationType?: string) {
```

Comment:
Why is this optional? Is it possible to refine this before calling these functions?

Reply:
>Why is this optional? Is it possible to refine this before calling these functions?

This is optional because it comes from the URL. We had a function that refined it, but we decided to remove it on another PR a few weeks ago.